### PR TITLE
examples: Add shared node/yarn recipe/s

### PR DESCRIPTION
**Pull request by phlax**
This is a breakout from #31534 but can be useful for a bunch of existing sandboxes also

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @htuch

File: examples/shared/node/Dockerfile [hunk 1]
```diff
@@ -9,3 +9,23 @@ ENV NODE_APP_PATH "/app/${NODE_APP_NAME}"
...
 # Dont use exec form to interpolate correctly
 CMD node $NODE_APP_PATH
+
+
+FROM node-base as yarn-routed
+ARG SERVICE_PORT=3000
+ENV DEBIAN_FRONTEND=noninteractive \
+    SERVICE_PORT=$SERVICE_PORT
+COPY --chmod=755 ./scripts/entrypoint.sh /entrypoint.sh
+COPY --chmod=755 ./scripts/build.sh /usr/local/bin/build.sh
```
Comment (phlax) [inline examples/shared/node/Dockerfile:19]:
```suggestion
COPY --chmod=755 ./scripts/build-routed.sh /usr/local/bin/build.sh
```

File: examples/shared/node/scripts/build.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+yarn
+yarn build
+
+BASE=$(yq -c < ../xds/lds.tmpl.yml)
+ASSETS=$(find dist -type f)
```
Comment (phlax) [inline examples/shared/node/scripts/build.sh:7]: rm

**Review by wbpcode — APPROVED** [review]
LGTM. Thank

**Pull request merged**